doc/port-si/port-si-per-speed.md

```diff
@@ -0,0 +1,267 @@
+regn_bfm1p          = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; voltage regulator to pre output buffer PMOS side
+regn_bfm1n          = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; voltage regulator to pre output buffer NMOS side
+```
+
+Here is the table to map the fields and SAI attributes:
+
+| Parameter     |              sai_port_attr_t                      |
+|:-------------:|:-------------------------------------------------:|
+| ob_m2lp	      |     SAI_PORT_SERDES_ATTR_TX_PAM4_RATIO            |
+| ob_alev_out   |     SAI_PORT_SERDES_ATTR_TX_OUT_COMMON_MODE       |
+| obplev        |     SAI_PORT_SERDES_ATTR_TX_PMOS_COMMON_MODE      |
+| obnlev        |     SAI_PORT_SERDES_ATTR_TX_NMOS_COMMON_MODE      |
+| regn_bfm1p    |     SAI_PORT_SERDES_ATTR_TX_PMOS VLTG_REG         |
+| regn_bfm1n    |     SAI_PORT_SERDES_ATTR_TX_NMOS VLTG_REG         |
+
+
+## The input
+
+SI parameters input comes as an excel file, received from PHY, and structured as follows:
+
+![Alt text](<input excel snippet-1.png>)
+
+
+Each pair of platform (NDR, HDR, XDR) and device (4700, 5600) has its own dedicated excel file.</br>
+This file contains both SerDes SI parameters and module SI parameters.</br>
+To ensure that this data is transmitted properly to SDK, the Excel file will be parsed and transformed into the media_settings.json file.</br>
+The media_settings.json file will be located at '/usr/share/sonic/device/[device_type]/media_settings.json' and will follow this format:
+
+![Alt text](<json template-1.png>)
+
+
+## How are we going to use this json?
+
+The flow of using of this json will be referred to as the **_Notify-Media-setting-Process_**:
+
+![Notify Media Setting Process](Notify-Media-Setting-Process.drawio.svg)
+
+
+During the **_Notify-Media-setting-Process_** two things occur:
+
+1. STATE_DB is updated with sfp, dom and pm data.
+2. The APP_DB is updated with the connected module SI parameters. This is achieved through the notify_media_settings() function, which uses the pre-parsed media_settings.json file to write its contents to APP_DB: First, a key is composed for performing the JSON lookup, based on the media type, specification compliance, length, and lane speed. Then, the lookup is performed using that key, and the relevant data is extracted from the JSON and stored in APP_DB.
+
+    Before transitioning to independent mode, SERDES SI parameters were never applied because the media_settings.json file was not found on the platform. In the new mode, media_settings.json will be present on every platform, but it will only be utilized when it contains an entry for a plugged-in CMIS module (QSFP-DD, OSFP). If it does not have a suitable entry for the connected module, nothing will be written to APP_DB to maintain backward compatibility with the legacy flow.
```

Comment:
A note to mention here is if for some reason a CMIS cable is replaced by non CMIS cable, the settings applied for the previous CMIS cable will still be in place (as there is no remove flow) and hence the new non CMIS module might not function properly.

Reply:
Remove flow can be found in chassis code